jmix-security/security-flowui/src/main/java/io/jmix/securityflowui/action/AssignToUsersAction.java

```diff
@@ -156,13 +156,13 @@ protected void openDialog() {
 
     protected void configureViewLoader(DataLoader loader) {
         List<String> excludedUsernames = dataManager.load(RoleAssignmentEntity.class)
-                .condition(PropertyCondition.equal(ROLE_CODE_PROPERTY, selectedItem.getCode()))
+                .condition(UIConditions.equal(ROLE_CODE_PROPERTY, selectedItem.getCode()))
```

Comment:
Why do we need to create a separate class UIConditions? Why not to simply add the `.skipNullOrEmpty()` to existing PropertyCondition builder invocation? E.g. `PropertyCondition.equal(ROLE_CODE_PROPERTY, selectedItem.getCode())` -> `PropertyCondition.equal(ROLE_CODE_PROPERTY, selectedItem.getCode()).skipNullOrEmpty()`. What are the benefits of using a new class that fully repeats condition builders?

Reply:
First of all it improves code readability, especially when condition created in the middle of complex code, e.g.
this part of code:
```java
 List<String> excludedUsernames = dataManager.load(RoleAssignmentEntity.class)
                .condition(UIConditions.equal(ROLE_CODE_PROPERTY, selectedItem.getCode()))
                .list()
                .stream()
                .map(RoleAssignmentEntity::getUsername)
                .collect(Collectors.toUnmodifiableList());
```
looks better than this:
```java
 List<String> excludedUsernames = dataManager.load(RoleAssignmentEntity.class)
                .condition(PropertyCondition.equal(ROLE_CODE_PROPERTY, selectedItem.getCode())
                        .skipNullOrEmpty())
                .list()
                .stream()
                .map(RoleAssignmentEntity::getUsername)
                .collect(Collectors.toUnmodifiableList());
```
or:
```java
  return UIConditions.jpqlCondition("e.parent = :parent", null, ParamsMap.of("parent", parentDocument));
```
vs.
```java
  return JpqlCondition.jpqlCondition("e.parent = :parent", null, ParamsMap.of("parent", parentDocument))
                    .skipNullOrEmpty();
```
also it allows to preserve space and write some statements in one line instead of splitting it.

Second reason: because of code simplification and atomicity it is more error-prone during code refactoring and development. Otherwise I afraid that `.skipNullOrEmpty()` part may be lost or forgotten.

Reply:
For me, "code readability" is when you look at the code and understand what it does, not when it takes less space on the screen. 

When I look at:

```
PropertyCondition.equal(ROLE_CODE_PROPERTY, selectedItem.getCode())
```

vs

```
PropertyCondition.equal(ROLE_CODE_PROPERTY, selectedItem.getCode()).skipNullOrEmpty()
```
then I immediately understand the difference between two methods.

When I look at

```
PropertyCondition.equal(ROLE_CODE_PROPERTY, selectedItem.getCode())
```

vs 

```
UIConditions.equal(ROLE_CODE_PROPERTY, selectedItem.getCode())
```

then for me it's harder to get the difference.

Also, for me it's always better to have one entry point to condition creation that having many of them without the clear reason. Remembering the rule: "If I want to create a condition then I should look at the static factory method in its class" is much simpler than "If I want these conditions then I need condition class, if I want that condition, then I need a separate UICondition" class.

Third, I'm not sure that many users except us will create skippable property conditions programmatically.

Let's try to ask a third opinion about this.

Reply:
Final Decision: remove `UiConditions` and use `.skipNullOrEmpty()` in each place